cpp/merkletree/compact_merkle_tree.cc

```diff
@@ -21,7 +21,7 @@ CompactMerkleTree::CompactMerkleTree(SerialHasher* hasher)
 
 CompactMerkleTree::CompactMerkleTree(MerkleTree& model, SerialHasher* hasher)
     : MerkleTreeInterface(),
-      tree_(std::max(0L, int64_t(model.LevelCount()) - 1)),
+      tree_(std::max(int64_t(0), int64_t(model.LevelCount()) - 1)),
```

Comment:
I can't do that as it causes one of the tree tests to fail with a
std::bad_alloc.

Martin

On 14 August 2015 at 13:57, Pierre Phaneuf <notifications@github.com> wrote:

> In cpp/merkletree/compact_merkle_tree.cc
> <https://github.com/google/certificate-transparency/pull/866#discussion_r37073293>
> :
>
> > @@ -21,7 +21,7 @@ CompactMerkleTree::CompactMerkleTree(SerialHasher* hasher)
> >
> >  CompactMerkleTree::CompactMerkleTree(MerkleTree& model, SerialHasher* hasher)
> >      : MerkleTreeInterface(),
> > -      tree_(std::max(0L, int64_t(model.LevelCount()) - 1)),
> > +      tree_(std::max(int64_t(0), int64_t(model.LevelCount()) - 1)),
>
> I think you could avoid this (and the C-style rando-casts) by doing this?
>
> std::max<int64_t>(0, model.LevelCount() - 1)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/certificate-transparency/pull/866/files#r37073293>
> .
>